neuro_san/internals/graph/activations/abstract_class_activation.py

```diff
@@ -225,9 +228,14 @@ async def attempt_invoke(self, coded_tool: CodedTool, arguments: Dict[str, Any],
         """
         retval: Any = None
 
+        tool_args: Dict[str, Any] = arguments.copy()
+        # Need to remove class references from the args that will not transfer over the wire
+        if "progress_reporter" in tool_args:
+            del tool_args["progress_reporter"]
```

Comment:
Need to remove instances in args that will not transfer over the wire for the AgentMessage below.